src/css.js

```diff
@@ -336,16 +333,23 @@ jQuery.each( [ "height", "width" ], function( i, name ) {
 		},
 
 		set: function( elem, value, extra ) {
-			var styles = extra && getStyles( elem );
-			return setPositiveNumber( elem, value, extra ?
-				augmentWidthOrHeight(
+			var matches = rcssNum.exec( value ),
```

Comment:
rcssNum allows leading `+=` and `-=`, which is invalid here. We should either keep `rnumsplit`, or verify `&& !matches[ 1 ]` in the condition below.